src/ContentRepository/Packaging/Steps/DeleteContentType.cs

```diff
@@ -0,0 +1,413 @@
+                    .Where(c => c.AllowedChildTypeNames != null && c.AllowedChildTypeNames.Intersect(names).Any())
+                    .ToArray();
+        }
+        private ReferenceFieldSetting[] GetContentTypesWhereTheyAreAllowedInReferenceField(string[] names)
+        {
+            return ContentType.GetContentTypes().SelectMany(t => t.FieldSettings)
+                .Where(f => f is ReferenceFieldSetting).Cast<ReferenceFieldSetting>()
+                .Where(r => r.AllowedTypes != null && r.AllowedTypes.Intersect(names).Any())
+                .Distinct()
+                .ToArray();
+        }
+        private Dictionary<string, string> GetContentPathsWhereTheyAreAllowedChildren(string[] names)
+        {
+            var result = new Dictionary<string, string>();
+
+            var whereClausePart = string.Join(Environment.NewLine + "    OR" + Environment.NewLine,
+                names.Select(n =>
+                    $"    (t.Value like '{n}' OR t.Value like '% {n} %' OR t.Value like '{n} %' OR t.Value like '% {n}')"));
+
+            // testability: the first line is recognizable for the tests.
+            var sql = $"-- GetContentPathsWhereTheyAreAllowedChildren: [{string.Join(", ", names)}]" +
+                      Environment.NewLine;
+            sql += @"SELECT n.Path, t.Value FROM TextPropertiesNVarchar t
+	JOIN SchemaPropertyTypes p ON p.PropertyTypeId = t.PropertyTypeId
+	JOIN Versions v ON t.VersionId = v.VersionId
+	JOIN Nodes n ON n.NodeId = v.NodeId
+WHERE p.Name = 'AllowedChildTypes' AND (
+" + whereClausePart + @"
+)
+UNION ALL
+SELECT n.Path, t.Value FROM TextPropertiesNText t
+	JOIN SchemaPropertyTypes p ON p.PropertyTypeId = t.PropertyTypeId
+	JOIN Versions v ON t.VersionId = v.VersionId
+	JOIN Nodes n ON n.NodeId = v.NodeId
+WHERE p.Name = 'AllowedChildTypes' AND (
+" + whereClausePart + @"
+)
+";
+
+            using (var cmd = DataProvider.CreateDataProcedure(sql))
+            {
+                cmd.CommandType = CommandType.Text;
+                using (var reader = cmd.ExecuteReader())
+                    while (reader.Read())
+                        result.Add(reader.GetString(0), reader.GetString(1));
+            }
+
+            return result;
+        }
+
+        private RelatedSensitiveFolders GetRelatedFolders(string[] names)
+        {
+            var result = ContentQuery.Query(ContentRepository.SafeQueries.TypeIsAndName, QuerySettings.AdminSettings,
+                "Folder", names);
+
+            var apps = new List<string>();
+            var temps = new List<string>();
+            var views = new List<string>();
+            var tempsPrefix = RepositoryStructure.ContentTemplateFolderPath.ToLowerInvariant();
+            foreach (var node in result.Nodes)
+            {
+                var path = node.Path.ToLowerInvariant();
+                if (path.Contains("/(apps)/"))
+                    apps.Add(path);
+                else if(path.StartsWith(tempsPrefix))
```

Comment:
Content templates can also be on lower levels (e.g. workspace or site level), not just in the root. So it would be better to test for the 'ContentTemplates' folder name too, not only the prefix.